### ISSUE TITLE
aria-keyshortcuts: Do multiple shortcuts define a set of alternatives, or a single sequence?

### ISSUE THREAD

**Issue by SyntaxColoring** [w3c/aria#2578]
# Describe your concern

If I write:

`aria-keyshortcuts="ctrl+a ctrl+b"`

Does that mean that, to activate the element, the user can press *either* Ctrl+A _or_ Ctrl+B? Or does it mean that they must press Ctrl+A _and then, in quick succession,_ press Ctrl+B?

I would think it's the former, but the MDN text ([permalink](https://github.com/mdn/content/blob/65f868c9be01412e35efc373ef0d8b4c9781e031/files/en-us/web/accessibility/aria/reference/attributes/aria-keyshortcuts/index.md)) has me second-guessing that (emphasis mine):

> Modifier keys must be listed first in each key combination. Possible key combinations include ` Control+P` or `Shift+Space` and `Q`. **If the full shortcut requires all three of these combinations in order, it would be written as `aria-keyshortcuts="Control+P Shift+Space Q`".**

I wasn't able to find an underlying source for that statement.

I *guess* the specification could be construed either way:

> The value of the `aria-keyshortcuts` attribute is a space-separated list of keyboard shortcuts that can be pressed to activate a command or textbox widget. The keys defined in the shortcuts represent the physical keys pressed and not the actual characters generated. Each keyboard shortcut consists of one or more tokens delimited by the plus sign ("+") representing zero or more modifier keys and exactly one non-modifier key that must be pressed simultaneously to activate the given shortcut.

# Link to the version of the specification or documentation you were looking at at.

Link to documentation: https://www.w3.org/TR/wai-aria-1.3/#aria-keyshortcuts

Does the issue exists in [the editors draft](https://w3c.github.io/aria/) (the editors draft is the most recent draft of the specification)? Yes

**Comment by front-endian** [comment]
Howdy @SyntaxColoring! As the standard is currently written, I believe it implies that it is a space separated list of shortcut**s** (plural), especially as that is a more common case. However, I agree that it is unclear and someone would also have to dig into how various accessibility APIs and assistive technologies present this information.

**Do you have a particular use case and/or examples that you can share** of shortcuts that require a sequence of multiple key combinations to activate? Is there a motivating example that caused you to open this issue?

I am aware of very few in real world applications that use sequences of keypresses like that for shortcuts, and they tend to violate [WCAG 2.2.1 (Timing Adjustable)](https://www.w3.org/TR/WCAG22/#timing-adjustable).

**Comment by SyntaxColoring** [comment]
> Do you have a particular use case and/or examples that you can share of shortcuts that require a sequence of multiple key combinations to activate?

I think the only times I've personally seen "sequence shortcuts" have been in text editors. e.g. in Sublime Text, opening/closing the sidebar is **Ctrl+K, Ctrl+B.** In Emacs, it's ubiquitous. *Maybe* I've also seen it in games.

> I am aware of very few in real world applications that use sequences of keypresses like that for shortcuts, and they tend to violate [WCAG 2.2.1 (Timing Adjustable)](https://www.w3.org/TR/WCAG22/#timing-adjustable).

Agreed. Though, someone else just reminded me that the sequences don't necessarily have to be timing-sensitive. Like, in Emacs, the shortcut sequence for quitting is **Ctrl+X, Ctrl+C**, and you can take as long as you like between the two.

> Is there a motivating example that caused you to open this issue?

My motivating example is kind of the opposite. I have a redo button that can be triggered by **Ctrl+Shift+Z** *or* **Ctrl+Y.** If I list both options, I want them to be understood as alternatives, not as a sequence. The MDN passage I quoted implied that it would be understood as a sequence, which is surprising to me and not what I want; but I couldn't find a source for it, so I'm hoping for clarification here on whether it's even true.

**Comment by spectranaut** [comment]
Discussed in today's meeting: https://www.w3.org/2025/09/18-aria-minutes.html#402f

**Comment by SyntaxColoring** [comment]
Here's an example of sequences used in Google Drive:

<img width="321" height="331" alt="Google Drive's 'New' dropdown. A few buttons have labels indicating keyboard shortcuts: '^C then F' for 'New folder', '^C then U' for 'File upload', and '^C then I' for 'Folder upload'." src="https://github.com/user-attachments/assets/c4933d10-b1c8-40d8-86e5-db8c8f92e368" />
